vpr/src/base/vpr_context.h

```diff
@@ -383,14 +383,48 @@ struct PackingMultithreadingContext : public Context {
  * or related placer algorithm state.
  */
 struct PlacementContext : public Context {
-    ///@brief Clustered block placement locations
-    vtr::vector_map<ClusterBlockId, t_block_loc> block_locs;
+  private:
+    /**
+     * Determines if place_loc_vars_ can be accessed by calling getter methods.
+     * This flag should be set to false at the beginning of the placement stage,
+     * and set to true at the end of placement. This ensures that variables that
+     * are subject to change during placement are kept local to the placement stage.
+     */
+    bool loc_vars_are_accessible_ = true;
+
+    /**
+     * @brief Stores block location information, which is subject to change during the
```

Comment:
Maybe say "persistent block location information". "This information is valid outside of placement optimization algorithms but it doesn't reflect the placement changes that are in progress by an optimizer such as the annealer. place_loc_vars_ is used to communicate placement state between major optimizers or from placement to routing, etc."